OpenKh.Imaging/Tm2.cs

```diff
@@ -341,18 +341,86 @@ private Tm2(Stream stream, Picture picture)
             ImageDataHelpers.InvertRedBlueChannels(_imageData, Size, PixelFormat);
         }
 
+        private Tm2(byte[] buffer, byte[] clut, Picture picture, PixelFormat pixFormat)
+        {
+            _imageFormat = picture.PictureFormat;
+            _mipMapCount = picture.MipMapCount;
+            _imageType = 5;
+            _clutType = picture.ClutType;
+            _gsTex0 = picture.GsTex0;
+            _gsTex1 = picture.GsTex1;
+            _gsReg = picture.GsRegs;
+            _gsPal = picture.GsClut;
+            Size = new Size(picture.Width, picture.Height);
+
+            if (picture.MipMapCount > 1)
+            {
+                _mipmap = new MipMap();
+                throw new NotImplementedException("Mipmaps are not currently supported.");
+            }
+
+            // picture.ClutSize is not valid for KH2 map radar.
+            var clutDataSize = 4 * picture.ClutColorCount;
+
+            _imageData = buffer;
+            _clutData = clut;
+            if (IsClutSwizzled)
+                _clutData = SortClut(_clutData, ClutFormat, picture.ClutColorCount);
+
+            ImageDataHelpers.InvertRedBlueChannels(_imageData, Size, pixFormat);
+        }
+
         public static bool IsValid(Stream stream) =>
             stream.SetPosition(0).ReadInt32() == MagicCode &&
             stream.Length >= HeaderLength;
 
-        public static IEnumerable<Tm2> Read(Stream stream)
+        public static Tm2 Create(byte[] buffer, short width, short height)
+        {
+            byte[] buff = buffer;
+            byte[] clut = new byte[1024];
+            Array.Resize(ref buff, buff.Length - 1024);
+            Array.Copy(buffer, buffer.Length - 1024, clut, 0, 1024);
+            Picture pic = new Picture();
+            pic.Width = width;
+            pic.Height = height;
+            pic.MipMapCount = 0;
+            return new Tm2(buff, clut, pic, PixelFormat.Indexed8);
+        }
+
+        public static Tm2 Create(PngImage image)
```

Comment:
Here you could have used a `IImageRead` instead of a `PngImage`

Reply:
I would have used it, but I don't know how. IImageRead is just an interface, so I don't know how I'm supposed to get any information other than `Size` and `PixelFormat`. Casting to PngImage?

Reply:
IImageRead has Size and PixelFormat. Those two properties you used from PngImage are actually from IImageRead interface. That way the `Tm2 Create` function can generate TM2 files from any other supported image formats in OpenKH!

Reply:
I just changed it to use IImageRead. Should be fine now, then.

Reply:
I think you have to create a new PR for that :'D

Reply:
Sorry for missing this detail. I will ensure that it does not happen again ^^'